sample-compose/domain/src/main/java/co/nimblehq/sample/compose/domain/repository/AppPreferencesRepository.kt

```diff
@@ -0,0 +1,11 @@
+package co.nimblehq.sample.compose.domain.repository
+
+import kotlinx.coroutines.flow.Flow
+
+interface AppPreferencesRepository {
+
+    fun isFirstTimeLaunchPreferences(): Flow<Boolean>
+
+    suspend fun updateFirstTimeLaunchPreferences(isFirstTimeLaunch: Boolean)
```

Comment:
I think we should not name these APIs with `Preferences` postfix as the wrapper is an `AppPreferences` already. This is the same when we define a Repository and all API functions inside not flowing with a `Repository` postfix.

Plus, `isFirstTimeLaunch` is not a `Preferences` object like the [Google example](https://developer.android.com/codelabs/android-preferences-datastore#5) because they have `UserPreferences` class as a wrapper.

```suggestion
    fun isFirstTimeLaunch(): Flow<Boolean>

    suspend fun updateFirstTimeLaunch(isFirstTimeLaunch: Boolean)
```